02-peer-protocol.md

```diff
@@ -270,6 +269,7 @@ The receiving node MUST fail the channel if:
   - `funding_pubkey`, `revocation_basepoint`, `htlc_basepoint`, `payment_basepoint`, or `delayed_payment_basepoint`
 are not valid secp256k1 pubkeys in compressed format.
   - `dust_limit_satoshis` is greater than `channel_reserve_satoshis`.
+  - `dust_limit_satoshis` is smaller than `354 satoshis` (see [BOLT 3](03-transactions.md#dust-limits)).
```

Comment:
I think this should only apply for `option_shutdown_anysegwit` and otherwise allow a lower bound of `330 satoshis`.

Reply:
> I think this should only apply for option_shutdown_anysegwit and otherwise allow a lower bound of 330 satoshis.

I disagree, because `option_shutdown_anysegwit` can be activated *after* the channel is opened.
For example, Alice and Bob open a channel without `option_shutdown_anysegwit` and set `dust_limit` to 330 sats.
Then later they both activate `option_shutdown_anysegwit` (because why not?).
If they didn't use an `upfront_shutdown_script`, they're allowed to use a segwit v1+ script when closing the channel, which can be an issue for dust limits below 354 sats.
So let's be safe and use 354 sats which is future-proof.

Reply:
I see, that makes sense.